aggregatestore/model/aggregatestore.go

```diff
@@ -75,7 +75,9 @@ func (r *AggregateStore) Save(ctx context.Context, aggregate eh.Aggregate) error
 
 	// Publish events if supported by the aggregate.
 	if publisher, ok := aggregate.(EventPublisher); ok && r.bus != nil {
-		for _, e := range publisher.EventsToPublish() {
+		events := publisher.EventsToPublish()
+		publisher.ClearEvents()
```

Comment:
I choose the same "guarantee" as the ES version, which is to clear all events before publishing, no matter what the result of publishing is. If an error happens here it will trickle to the command handler anyway, and should be handled by the client with a new request. There is a much broader issue about what to do with failing events in #181.

Reply:
Everything checks out. The old code crashed without my hack. Everything works ok with upstream master now. Thanks.